src/FSharpPlus.AspNetCore.Suave/Library.fs

```diff
@@ -98,43 +98,16 @@ module Filters=
   let OPTIONS (x : Http.Context) = response "OPTIONS" x
   let PATCH (x : Http.Context) = response "PATCH" x
   let path s = let path = implicit s in OptionT << fun (x : Http.Context) -> async.Return (if (path = x.request.Path) then Some x else None)
-  /// note: This implementation is smaller in scope than you'd find in Giraffe or Suave
-  /// I'd prefer to use some sort of library to turn path into a match expression
-  [<Obsolete("please use pathScan")>]
-  let inline pathRegex (path) (routeHandler : 'T -> WebPart<Context>) : WebPart<Context>=
-    let regex =Regex("^"+path+"$", RegexOptions.IgnoreCase)
-    let tryMatchInput = fun v->
-      let m = regex.Match v
-      if m.Success then tryParse m.Groups.[1].Value else None
-    fun (x : Http.Context) -> monad {
-      let pathValue = x.request.Path.Value
-      match tryMatchInput pathValue with
-      | Some args -> return! routeHandler args x
-      | None -> return! WebPart.fail x
-    }
-  /// note: This implementation is smaller in scope than you'd find in Giraffe or Suave
-  /// I'd prefer to use some sort of library to turn path into a match expression
-  [<Obsolete("please use pathScan")>]
-  let inline pathRegex2 (path) (routeHandler : 'T1 -> 'T2 -> WebPart<Context>) : WebPart<Context>=
-    let regex =Regex("^"+path+"$", RegexOptions.IgnoreCase)
-    let tryMatchInput = fun v->
-      let m = regex.Match v
-      if m.Success then (tryParse m.Groups.[1].Value, tryParse m.Groups.[2].Value) else None,None
-    fun (x : Http.Context) -> monad {
-      let pathValue = x.request.Path.Value
-      match tryMatchInput pathValue with
-      | Some args1, Some args2 -> return! routeHandler args1 args2 x
-      | _ -> return! WebPart.fail x
-    }
 
   let inline pathScan (path) (routeHandler) : WebPart<Context>=
     fun (x : Http.Context) ->
       if x.request.Path.HasValue then
```

Comment:
Is `x.request.Path` an option?
If so, you can do a single match, instead of if then + match:
`match x.request.Path >>= trySscanf with` and you'll save 3 lines of code.

Reply:
It's `Nullable<>`

Reply:
I'm wrong, it's PathString